### PR TITLE
[SYCL][ESIMD][ABI-Break] Remove deprecated APIs

### PR DESCRIPTION
**Pull request by turinevgeny**

### CODE REVIEW

**Review by sarnex — APPROVED** [review]
nice cleanup!

File: sycl/test-e2e/ESIMD/lsc/lsc_usm_2d.cpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ int main() {
                            cache_hint::uncached>(
                 input, (data_width * sizeof(int)) - 1, data_height - 1,
                 (data_pitch * sizeof(int)) - 1, x, y);
-            auto data = lsc_load2d<int, Width, Height, NumBlocks, false, false,
+            auto data = lsc_load_2d<int, Width, Height, NumBlocks, false, false,
```
Comment (v-klochkov) [inline sycl/test-e2e/ESIMD/lsc/lsc_usm_2d.cpp:64]: Shouldn't the call at L60 also be changed? lsc_prefetch2d

Comment (turinevgeny) [reply sycl/test-e2e/ESIMD/lsc/lsc_usm_2d.cpp:64]: Yes.

**Review by v-klochkov — APPROVED** [review]
Looks good!

**Comment by v-klochkov** [comment]
@turinevgeny - please fix clang-format issues

**Comment by turinevgeny** [comment]
> @turinevgeny - please fix clang-format issues

Done.

**Pull request merged**